### PR TITLE
Kdesktop 1343 enhance handling of file access error in executor worker

### PR DESCRIPTION
**Pull request by herve-er**

### CODE REVIEW

File: src/libsyncengine/syncpal/syncpal.h [hunk 1]
```diff
@@ -257,8 +257,10 @@ class SYNCENGINE_EXPORT SyncPal : public std::enable_shared_from_this<SyncPal> {
         virtual void refreshTmpBlacklist();
         virtual void removeItemFromTmpBlacklist(const NodeId &nodeId, ReplicaSide side);
         virtual void removeItemFromTmpBlacklist(const SyncPath &relativePath);
-
         ExitInfo handleAccessDeniedItem(const SyncPath &relativePath, ExitCause cause = ExitCause::FileAccessError);
+        ExitInfo handleAccessDeniedItem(const SyncPath &relativePath, std::shared_ptr<Node> &remoteBlacklistedNode,
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/syncpal/syncpal.h:261]: We usually follow the order, `local` then `remote`.

File: src/libsyncengine/propagation/executor/executorworker.cpp [hunk 2]
```diff
@@ -2627,15 +2648,15 @@ ExitInfo ExecutorWorker::removeDependentOps(SyncOpPtr syncOp) {
 
         if (localNode && localNode2 && (localNode->isParentOf(localNode2))) {
             LOGW_SYNCPAL_DEBUG(_logger, L"Removing " << syncOp2->type() << L" operation on " << SyncName2WStr(nodeName)
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/propagation/executor/executorworker.cpp:2650]:
```suggestion
            LOGW_SYNCPAL_DEBUG(_logger, L"Removing " << syncOp2->type() << L" operation on " << Utility::formatSyncName(nodeName)
```

File: src/libsyncengine/propagation/executor/executorworker.cpp [hunk 3]
```diff
@@ -2627,15 +2648,15 @@ ExitInfo ExecutorWorker::removeDependentOps(SyncOpPtr syncOp) {
...
-                                                     << L" because it depends on " << syncOp->type() << L" operation on "
+                                                     << L" because it depends on " << opType << L" operation on "
                                                      << SyncName2WStr(localNode->name()) << L" wich failed.");
             dependentOps.push_back(opId);
             continue;
         }
 
         if (remoteNode && remoteNode2 && (remoteNode->isParentOf(remoteNode2))) {
             LOGW_SYNCPAL_DEBUG(_logger, L"Removing " << syncOp2->type() << L" operation on " << SyncName2WStr(nodeName)
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/propagation/executor/executorworker.cpp:2658]:
```suggestion
            LOGW_SYNCPAL_DEBUG(_logger, L"Removing " << syncOp2->type() << L" operation on " << Utility::formatSyncName(nodeName)
```

File: src/libsyncengine/propagation/executor/executorworker.cpp [hunk 4]
```diff
@@ -2577,6 +2583,16 @@ ExitInfo ExecutorWorker::handleOpsAlreadyExistError(SyncOpPtr syncOp, ExitInfo o
             _syncPal->setRestart(true);
             return removeDependentOps(syncOp);
         }
+
+        // Check if we got the read right on the local item
+        IoError ioError = IoError::Unknown;
+        bool exist = false;
+        IoHelper::checkIfPathExists(_syncPal->localPath() / relativeLocalPath, exist, ioError);
+        if (ioError == IoError::AccessDenied) {
+            LOGW_DEBUG(_logger, Utility::formatSyncPath(relativeLocalPath) << L"has no read access, converting already exist error in file access error.");
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/propagation/executor/executorworker.cpp:2592]:
```suggestion
            LOGW_DEBUG(_logger, Utility::formatSyncPath(relativeLocalPath) << L"has no read access, converting 'already exist error' to 'file access error'.");
```

File: src/libsyncengine/syncpal/syncpal.h (hunk 1 above)
Comment (herve-er) [reply src/libsyncengine/syncpal/syncpal.h]: Done here: [c2a7256](https://github.com/Infomaniak/desktop-kDrive/pull/375/commits/c2a72567636c29932e2f6dbbdc9fddaa9c77ba36)

File: src/libsyncengine/propagation/executor/executorworker.cpp (hunk 4 above)
Comment (herve-er) [reply src/libsyncengine/propagation/executor/executorworker.cpp]: I am just waiting the PR including toString(ExitInfo) and I will directly log an exit info (https://github.com/Infomaniak/desktop-kDrive/pull/396)

**Review by ChristopheLarchier — APPROVED** [review]

File: src/libsyncengine/propagation/executor/executorworker.cpp (hunk 4 above)
Comment (herve-er) [reply src/libsyncengine/propagation/executor/executorworker.cpp]: Done here: [aec13c9](https://github.com/Infomaniak/desktop-kDrive/pull/375/commits/aec13c96a0c87cde95baee09ee6d50ad02f486d1)

**Review by ChristopheLarchier — APPROVED** [review]

**Pull request merged**